Assets/Scripts/Runtime/HPSystem/PlayerHealth.cs

```diff
@@ -0,0 +1,56 @@
+using Assets.Scripts.Runtime.Character;
+using StarterAssets;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerHealth : Health
+{
+    [SerializeField] private ThirdPersonController playerThirdPersonController;
+    [SerializeField] private Vector3 _respawnPosition;
+    [SerializeField] private Hero _hero;
+
+    protected override void OnDeath()
+    {
+        if(!isAlive)
+        {
+            playerThirdPersonController.enabled = false;
+        }
+    }
+    
+    protected override void Respawning()
+    {
+        transform.position = _respawnPosition;
+        Physics.SyncTransforms();
+        if (!playerThirdPersonController.enabled)
+        {
+            playerThirdPersonController.enabled = true;
+        }
+        if (_hero.GetMinions.Count > 0)
+        {
+            foreach (Minion minion in _hero.GetMinions)
+            {
+                minion.gameObject.transform.position = _respawnPosition;
+                minion.GoToPostion(_respawnPosition);
+                minion.StopAllCoroutines();
+            }
+        }
```

Comment:
this wont work after merge
generally this should call something like
`_hero.Respawn()`
and Respawn would do something similar 
but after refactor i think this will just work,
maybe it should Invoke some event
for `MinionStateFollowPlayer` to update `_lastHeroLocation`